doc/sphinx/addendum/implicit-coercions.rst

```diff
@@ -284,138 +272,137 @@ Examples
+    Parameter f : forall n:nat, C n -> D (S n) true.
+    Coercion f : C >-> D.
+    Parameter g : forall (n:nat) (b:bool), D n b -> E b.
+    Coercion g : D >-> E.
+    Parameter c : C 0.
+    Parameter T : E true -> nat.
+    Check (T c).
+    Set Printing Coercions.
+    Check (T c).
+    Unset Printing Coercions.
+
+  In the case of functional arguments, we use the monotonic rule of
+  sub-typing. To coerce :g:`t : forall x : A, B` towards
+  :g:`forall x : A', B'`, we have to coerce ``A'`` towards ``A`` and ``B``
+  towards ``B'``. An example is given below:
+
+  .. coqtop:: all
+
+    Parameters (A B : Set) (h : A -> B).
+    Coercion h : A >-> B.
+    Parameter U : (A -> E true) -> nat.
+    Parameter t : B -> C 0.
+    Check (U t).
+    Set Printing Coercions.
+    Check (U t).
+    Unset Printing Coercions.
+
+  Remark the changes in the result following the modification of the
+  previous example.
+
+  .. coqtop:: all
+
+    Parameter U' : (C 0 -> B) -> nat.
+    Parameter t' : E true -> A.
+    Check (U' t').
+    Set Printing Coercions.
+    Check (U' t').
+    Unset Printing Coercions.
+
+.. example:: Coercion to a type
+
+  An assumption ``x:A`` when ``A`` is not a type, is ill-typed.  It is
+  replaced by ``x:A'`` where ``A'`` is the result of the application to
+  ``A`` of the coercion path between the class of ``A`` and
+  ``Sortclass`` if it exists.  This case occurs in the abstraction
+  :g:`fun x:A => t`, universal quantification :g:`forall x:A,B`, global
+  variables and parameters of (co)inductive definitions and
+  functions. In :g:`forall x:A,B`, such a coercion path may also be applied
+  to ``B`` if necessary.
+
+  .. coqtop:: all
+
+    Parameter Graph : Type.
+    Parameter Node : Graph -> Type.
+    Coercion Node : Graph >-> Sortclass.
+    Parameter G : Graph.
+    Parameter Arrows : G -> G -> Type.
+    Check Arrows.
+    Parameter fg : G -> G.
+    Check fg.
+    Set Printing Coercions.
+    Check fg.
+    Unset Printing Coercions.
+
+.. example:: Coercion to a function
+
+  ``f a`` is ill-typed because ``f:A`` is not a function. The term
+  ``f`` is replaced by the term obtained by applying to ``f`` the
+  coercion path between ``A`` and ``Funclass`` if it exists.
+
+  .. coqtop:: all
+
+    Parameter bij : Set -> Set -> Set.
+    Parameter ap : forall A B:Set, bij A B -> A -> B.
+    Coercion ap : bij >-> Funclass.
+    Parameter b : bij nat nat.
+    Check (b 0).
+    Set Printing Coercions.
+    Check (b 0).
+    Unset Printing Coercions.
+
+.. _example-identity-coercion:
+
+.. example:: Identity coercions.
+
+  .. coqtop:: all
+
+    Definition fct := nat -> nat.
+    Parameter incr_fct : Set.
+    Parameter fct_of_incr_fct : incr_fct -> fct.
+    Fail Coercion fct_of_incr_fct : incr_fct >-> Funclass.
+    Coercion fct_of_incr_fct : incr_fct >-> fct.
+    Parameter f' : incr_fct.
+    Check f' : fct.
+    Fail Check f' 0.
+    Identity Coercion Id_fct_Funclass : fct >-> Funclass.
+    Check f' 0.
```

Comment:
```suggestion
  .. coqtop:: in

    Definition fct := nat -> nat.
    Parameter incr_fct : Set.
    Parameter fct_of_incr_fct : incr_fct -> fct.
    
 .. coqtop:: all   
 
    Fail Coercion fct_of_incr_fct : incr_fct >-> Funclass.
    
 .. coqtop:: in
 
    Coercion fct_of_incr_fct : incr_fct >-> fct.
    Parameter f' : incr_fct.
    
 .. coqtop:: all
 
    Check f' : fct.
    Fail Check f' 0.
    Identity Coercion Id_fct_Funclass : fct >-> Funclass.
    Check f' 0.
```

While a bit tedious, I try to remove the boring parts of examples to make them more readable, e.g.

![image](https://user-images.githubusercontent.com/1253341/158275627-70262c6f-9ff5-47da-8c13-fc0442e84f2d.png)